Error while checking build/O2/fullCI for 2765853de6eb33c0bfe7f6bd113bddc92ad1b6f2 at 2024-01-15 16:56:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile digi.log
digi.log-[45747:internal-dpl-ccdb-backend]: [15:56:29][FATAL] Unable to find object MFT/Calib/TimeDeadMap/1550600800000
digi.log:[45747:internal-dpl-ccdb-backend]: [15:56:29][ERROR] Exception caught: Fatal error 
[45747:internal-dpl-ccdb-backend]: [15:56:29][ERROR] CcdbDownloader finished transfer http://alice-ccdb.cern.ch/MFT/Calib/TimeDeadMap for 1550600800000 (agent_id: alissandra02-1705334175-lHPzLl) with http code: 404
[45747:internal-dpl-ccdb-backend]: [15:56:29][ERROR] File MFT/Calib/TimeDeadMap could not be retrieved. No more hosts to try.
[45747:internal-dpl-ccdb-backend]: [15:56:29][FATAL] Unable to find object MFT/Calib/TimeDeadMap/1550600800000
[45747:internal-dpl-ccdb-backend]: [15:56:29][ERROR] Exception caught: Fatal error 
[45747:internal-dpl-ccdb-backend]: [15:56:30][FATAL] Unhandled o2::framework::runtime_error reached the top of main of o2-sim-digitizer-workflow, device shutting down. Reason: Fatal error
[ERROR] Workflow crashed - PID 45747 (internal-dpl-ccdb-backend) did not exit correctly however it's not clear why. Exit code forced to 128.


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/5f1d6f204f94f5bdb9431191e549bb088bd1337f/slc8_x86-64/o2checkcode/1.0-local422/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12525/2765853de6eb33c0bfe7f6bd113bddc92ad1b6f2/build_O2_fullCI/pretty.html).

--- a/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
+++ b/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
@@ -105,7 +105,7 @@ class NoiseMap
   void applyProbThreshold(float t, long int n, float relErr = 0.2f, int minChipID = 0, int maxChipID = 24119)
   {
     // Remove from the maps all pixels with the firing probability below the threshold
-    // Apply the cut only for chips between minChipID and maxChipID (included).
+    // Apply the cut only for chips between minChipID and maxChipID (included)
     if (n < 1) {
       LOGP(alarm, "Cannot apply threshold with {} ROFs scanned", n);
       return;

--- a/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
+++ b/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
@@ -105,7 +105,7 @@ class NoiseMap
   void applyProbThreshold(float t, long int n, float relErr = 0.2f, int minChipID = 0, int maxChipID = 24119)
   {
     // Remove from the maps all pixels with the firing probability below the threshold
-    // Apply the cut only for chips between minChipID and maxChipID (included)
+    // Apply the cut only for chips between minChipID and maxChipID (included).
     if (n < 1) {
       LOGP(alarm, "Cannot apply threshold with {} ROFs scanned", n);
       return;

--- a/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
+++ b/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
@@ -169,31 +169,7 @@ class NoiseMap
   // Methods required by the calibration framework
   void print();
   void fill(const gsl::span<const CompClusterExt> data);
-  void merge(const NoiseMap* prev, bool sumstrobes = false)
-  {
-    if (prev == nullptr) {
-      return;
-    }
-    if (mNoisyPixels.size() != prev->size()) {
-      LOGP(error, "Trying to merge NoiseMap with different sizes: {} vs {}. Doing nothing.", mNoisyPixels.size(), prev->size());
-      return;
-    }
-    if (sumstrobes) {
-      mNumOfStrobes += prev->getNumberOfStrobes();
-    }
-
-    for (size_t chipID = 0; chipID < mNoisyPixels.size(); chipID++) {
-
-      const auto* prevChipMap = prev->getChipMap(chipID);
-
-      if (prevChipMap) {
-        // Iterate over previous map and add its entries
-        for (const auto& p : *prevChipMap) {
-          mNoisyPixels[chipID][p.first] = std::max(mNoisyPixels[chipID][p.first], p.second);
-        }
-      }
-    }
-  }
+  void merge(const NoiseMap* prev) {}
 
   const std::map<int, int>* getChipMap(int chip) const { return chip < (int)mNoisyPixels.size() ? &mNoisyPixels[chip] : nullptr; }
 

--- a/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/TimeDeadMap.h
+++ b/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/TimeDeadMap.h
@@ -40,12 +40,12 @@ class TimeDeadMap
   /// Destructor
   ~TimeDeadMap() = default;
 
-  void fillMap(unsigned long firstOrbit, std::vector<uint16_t> deadVect)
+  void fillMap(unsigned long firstOrbit, const std::vector<uint16_t>& deadVect)
   {
     mEvolvingDeadMap[firstOrbit] = deadVect;
   };
 
-  void fillMap(std::vector<uint16_t> deadVect)
+  void fillMap(const std::vector<uint16_t>& deadVect)
   {
     mStaticDeadMap = deadVect;
   }
@@ -56,25 +56,25 @@ class TimeDeadMap
     mStaticDeadMap.clear();
   }
 
-  void decodeMap(o2::itsmft::NoiseMap* noisemap)
+  void decodeMap(o2::itsmft::NoiseMap& noisemap)
   { // for static part only
     if (mMAP_VERSION != "3") {
       LOG(error) << "Trying to decode static part of deadmap version " << mMAP_VERSION << ". Not implemented, doing nothing.";
       return;
     }
     for (int iel = 0; iel < mStaticDeadMap.size(); iel++) {
-      uint16_t w = mStaticDeadMap.at(iel);
-      noisemap->maskFullChip(w & 0x7FFF);
+      uint16_t w = mStaticDeadMap[iel];
+      noisemap.maskFullChip(w & 0x7FFF);
       if (w & 0x8000) {
         for (int w2 = (w & 0x7FFF) + 1; w2 < mStaticDeadMap.at(iel + 1); w2++) {
-          noisemap->maskFullChip(w2);
+          noisemap.maskFullChip(w2);
         }
       }
     }
   }
 
-  void decodeMap(unsigned long orbit, o2::itsmft::NoiseMap* noisemap, bool includeStaticMap = true) // for time-dependent and (optionally) static part
-  {
+  void decodeMap(unsigned long orbit, o2::itsmft::NoiseMap& noisemap, bool includeStaticMap = true)
+  { // for time-dependent and (optionally) static part
 
     if (mMAP_VERSION != "3" && mMAP_VERSION != "4") {
       LOG(error) << "Trying to decode time-dependent deadmap version " << mMAP_VERSION << ". Not implemented, doing nothing.";
@@ -101,18 +101,18 @@ class TimeDeadMap
 
     for (int iel = 0; iel < closestVec.size(); iel++) {
       uint16_t w = closestVec.at(iel);
-      noisemap->maskFullChip(w & 0x7FFF);
+      noisemap.maskFullChip(w & 0x7FFF);
       if (w & 0x8000) {
         for (int w2 = (w & 0x7FFF) + 1; w2 < closestVec.at(iel + 1); w2++) {
-          noisemap->maskFullChip(w2);
+          noisemap.maskFullChip(w2);
         }
       }
     }
   };
 
-  std::string getMapVersion() { return mMAP_VERSION; };
+  std::string getMapVersion() const { return mMAP_VERSION; };
 
-  unsigned long getEvolvingMapSize() { return mEvolvingDeadMap.size(); };
+  unsigned long getEvolvingMapSize() const { return mEvolvingDeadMap.size(); };
 
   std::vector<unsigned long> getEvolvingMapKeys()
   {
@@ -126,6 +126,10 @@ class TimeDeadMap
 
   long getMapAtOrbit(unsigned long orbit, std::vector<uint16_t>& mmap)
   { // fills mmap and returns orbit - upper_bound
+    if (mEvolvingDeadMap.empty()) {
+      LOG(warning) << "Requested orbit " << orbit << "from an empty time-dependent map. Returning empty vector.";
+      return (long)orbit;
+    }
     auto closest = mEvolvingDeadMap.lower_bound(orbit);
     if (closest != mEvolvingDeadMap.end()) {
       mmap = closest->second;

--- a/Detectors/ITSMFT/common/simulation/include/ITSMFTSimulation/Digitizer.h
+++ b/Detectors/ITSMFT/common/simulation/include/ITSMFTSimulation/Digitizer.h
@@ -59,7 +59,7 @@ class Digitizer : public TObject
   o2::itsmft::DigiParams& getParams() { return (o2::itsmft::DigiParams&)mParams; }
   const o2::itsmft::DigiParams& getParams() const { return mParams; }
   void setNoiseMap(const o2::itsmft::NoiseMap* mp) { mNoiseMap = mp; }
-  void setDeadChannelsMap(o2::itsmft::NoiseMap* mp) { mDeadChanMap = mp; }
+  void setDeadChannelsMap(const o2::itsmft::NoiseMap* mp) { mDeadChanMap = mp; }
 
   void init();
 
@@ -137,7 +137,7 @@ class Digitizer : public TObject
   std::vector<o2::itsmft::ROFRecord>* mROFRecords = nullptr;               //! output ROF records
   o2::dataformats::MCTruthContainer<o2::MCCompLabel>* mMCLabels = nullptr; //! output labels
   const o2::itsmft::NoiseMap* mNoiseMap = nullptr;
-  o2::itsmft::NoiseMap* mDeadChanMap = nullptr;
+  const o2::itsmft::NoiseMap* mDeadChanMap = nullptr;
 
   ClassDefOverride(Digitizer, 2);
 };

--- a/Steer/DigitizerWorkflow/src/ITSMFTDigitizerSpec.cxx
+++ b/Steer/DigitizerWorkflow/src/ITSMFTDigitizerSpec.cxx
@@ -199,7 +199,12 @@ class ITSMFTDPLDigitizerTask : BaseDPLDigitizer
     if (matcher == ConcreteDataMatcher(mOrigin, "TimeDeadMap", 0)) {
       o2::itsmft::TimeDeadMap* timedeadmap = (o2::itsmft::TimeDeadMap*)obj;
       if (!timedeadmap->isDefault()) {
-        timedeadmap->decodeMap(mFirstOrbitTF, mDeadMap, true);
+        timedeadmap->decodeMap(mFirstOrbitTF, *mDeadMap, true);
+        static bool UpdateDone = false;
+        if (UpdateDone) {
+          LOGP(fatal, "Attempt to add time-dependent map to already modified static map");
+        }
+        UpdateDone = true;
         mDigitizer.setDeadChannelsMap(mDeadMap);
         LOG(info) << mID.getName() << " time-dependent dead map updated";
       } else {

